libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py

```diff
@@ -195,12 +187,20 @@ def _post(
         """Method for posting to the AI Foundation Model Function API."""
         self.last_inputs = {
             "url": invoke_url,
-            "headers": self.headers["call"],
+            # "headers": self.headers["call"],
             "json": self.payload_fn(payload),
             "stream": False,
         }
+        headers = {
```

Comment:
how about leaving `last_inputs` the same and passing call `session.post(**self._add_bearer(self.last_inputs))`

`_add_bearer` would do the appropriate checks for api_key and replacement.

this approach would reduce duplication across `session.post` call sites and allow `last_inputs` to include the `"Authorization": "Bearer {api_key}"` information.

Reply:
If I process in last_input then the authorization shows the api_key as it is changing inplace